internal/parse/parts.go

```diff
@@ -0,0 +1,155 @@
+package parse
+
+import "fmt"
+
+// queryPart is an interface to be implemented by all parts.
+type queryPart interface {
+	// String-like output for debugging purposes
+	String() string
+
+	// Translate part to its Sql equivalent
+	ToSql() string
+}
+
+// qualifiedExpression represents a qualified expression
+// qualified. Examples are:
+//	... column as &Person ... 	<--- "column" is unqualified
+//	... t.column as &Person.name	<--- "name" and "column" and qualified
+type qualifiedExpression struct {
+	qualifier string
+	name      string
+}
+
+// InputPart represents an expression used as input in the SDL query.
+// Examples:
+//	$Address
+//	$Address.postal_code
+type InputPart struct {
+	goType
+}
+
+// NewInputPart is a constructor that returns a pointer to an InputPart.
+func NewInputPart(typeName string, tagName string) (*InputPart, error) {
+	if typeName == "" && tagName != "" {
+		return nil, fmt.Errorf("malformed InputPart")
+	}
+	return &InputPart{
+		goType{
+			qualifiedExpression{qualifier: typeName,
+				name: tagName,
+			},
+		},
+	}, nil
+}
+
+func (ip *InputPart) String() string {
```

Comment:
Nitpick: "ip" reads as "IP" :)

I suggest using just "p" for all parts.

Reply:
Yep, you're right.